src/chainhook/helpers.ts

```diff
@@ -0,0 +1,104 @@
+import { OrdinalSatoshi } from '../api/util/ordinal-satoshi';
+import { logger } from '../logger';
+import { PgStore } from '../pg/pg-store';
+import { ChainhookPayloadCType } from './schemas';
+
+/**
+ * Process an `inscription_feed` event from chainhooks and saves inscription data to the DB.
+ * @param payload - Event payload
+ * @param db - DB
+ */
+export async function processInscriptionFeed(payload: unknown, db: PgStore): Promise<void> {
```

Comment:
It seems like all the sql writes inside this function should be performed inside a sql transaction

Reply:
I actually had this before but it created locking problems in postgres because we do both INSERT and UPDATE on location rows in order to mark the new inscription locations. I moved it over to atomic writes for each inscription movement to workaround this problem. What do you think?

Reply:
I think there could be two problems here:
* If the chainhook node performs another POST to `/chainhook/inscription_feed` while this function is still running, we'll have a race condition that could cause bugged writes. In the Stacks API we use p-queue to ensure writes finish before the next one comes in (so that could be fixed by doing a similar promise queue / unrelated to sql transactions).
* If this whole write block isn't in a sql transaction, then reads can come in the middle of this write and return buggy data.

The first issue seems critical because it could corrupt the db. The second issue seems like it would "only" result in bugs in the read path, so not the end of the world, but may still cause confusion and bugs for clients.

Reply:
I think the current set up works because Chainhooks is synchronous i.e. it sends one block at a time sequentially. However, I agree we should do this. What I'll do is separate the INSERTs from the UPDATEs and have it be an eventually consistent approach similar to the Stacks API.

I'll send you a separate PR for this because this one could be a bit large.